### PR TITLE
Alessio confirm upgrade

### LINKED ISSUE #258: Add support for 3d secure cards to upgrade
## Feature request
Add modal for payment confirm by secure 3d authentication to the upgrade plan

### Feature description
Some cards at the time of the upgrade went wrong because they need authentication at every payment.

@simoneBrancati 
at the 'stripe/customer/update/preview' endpoint i need this new data:
```
body: {
  data: {
    AmountPreview
    name
    email
    expiration
    cardNumber
  }
}

````

and for 'stripe/customer/update' endpoint' i need: 
```
body: {
  data: {
    status:
    paymentMethod 
    clientSecret
  }
}

````

@danielapetix  I need an update to the mockup for the upgrade / downgrade confirmation page.
The page must now contain the following info:
- Name card holder

- Email card holder

- Card Number (last 4 characters)

- Card expiration date

### PR DESCRIPTION
**Pull request by cpalessio96**
### All Submissions:

*   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
*   [x] Does your code has proper inline documentation? 

### Changes proposed in this Pull Request:

add support 3d secure card for upgrade plan 

Closes #258 .

### CODE REVIEW

**Comment by GiuseppeIuculano** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage decreased per file
===========================
- includes/Api/ApiPricingPlan.php  -2
         

Complexity increasing per file
==============================
- assets/payment/change_plan_script.js  1
         

Complexity decreasing per file
==============================
+ src/stripe/SubscriptionForm.js  -1
+ src/stripe/UpdateForm.js  -1
+ includes/Dashboard/PaymentTemplate.php  -1
+ assets/payment/payment_template.js  -1
         

Clones added
============
- assets/payment/settings_script.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/CTMobi/Wubtitle-Wordpress-Plugin/pullRequest?prid=5959027&bid=19536256)

**Comment by GiuseppeIuculano** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- assets/payment/change_plan_script.js  1
         

Complexity decreasing per file
==============================
+ assets/payment/payment_template.js  -1
+ includes/Dashboard/PaymentTemplate.php  -1
+ src/stripe/UpdateForm.js  -1
+ src/stripe/SubscriptionForm.js  -1
         

Clones added
============
- assets/payment/settings_script.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/CTMobi/Wubtitle-Wordpress-Plugin/pullRequest?prid=5959027&bid=19536256)

**Comment by GiuseppeIuculano** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage increased per file
===========================
+ includes/Api/ApiPricingPlan.php  3
         

Complexity increasing per file
==============================
- assets/payment/change_plan_script.js  1
         

Complexity decreasing per file
==============================
+ includes/Dashboard/PaymentTemplate.php  -1
+ assets/payment/payment_template.js  -1
+ src/stripe/SubscriptionForm.js  -1
+ src/stripe/UpdateForm.js  -1
         

Clones added
============
- assets/payment/settings_script.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/CTMobi/Wubtitle-Wordpress-Plugin/pullRequest?prid=5959027&bid=19536256)

File: assets/payment/change_plan_script.js [hunk 1]
```diff
@@ -13,10 +32,20 @@
...
+					) {
+						confirmPayment(
+							response.data.clientSecret,
+							response.paymentMethod
+						);
+					} else {
+						window.unonload = window.opener.redirectToCallback(
+							'notices-code=payment'
+						);
+						window.close();
```
Comment (HyprCoder) [inline assets/payment/change_plan_script.js]: This code is duplicated in line 13-16. Remove the duplication by creating a function

**Review by HyprCoder — CHANGES_REQUESTED** [review]

**Comment by GiuseppeIuculano** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- assets/payment/change_plan_script.js  1
         

Complexity decreasing per file
==============================
+ src/stripe/SubscriptionForm.js  -1
+ src/stripe/UpdateForm.js  -1
+ includes/Dashboard/PaymentTemplate.php  -1
+ assets/payment/payment_template.js  -1
         

Clones added
============
- assets/payment/settings_script.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/CTMobi/Wubtitle-Wordpress-Plugin/pullRequest?prid=5959027&bid=19536256)

**Review by HyprCoder — DISMISSED** [review]

**Comment by GiuseppeIuculano** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- assets/payment/change_plan_script.js  1
         

Complexity decreasing per file
==============================
+ src/stripe/UpdateForm.js  -1
+ includes/Dashboard/PaymentTemplate.php  -1
+ src/stripe/SubscriptionForm.js  -1
+ assets/payment/payment_template.js  -1
         

Clones added
============
- assets/payment/settings_script.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/CTMobi/Wubtitle-Wordpress-Plugin/pullRequest?prid=5959027&bid=19536256)

**Comment by GiuseppeIuculano** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- assets/payment/change_plan_script.js  1
         

Complexity decreasing per file
==============================
+ src/stripe/UpdateForm.js  -1
+ assets/payment/payment_template.js  -1
+ src/stripe/SubscriptionForm.js  -1
+ includes/Dashboard/PaymentTemplate.php  -1
         

Clones added
============
- assets/payment/settings_script.js  2
         

Clones removed
==============
+ includes/Api/ApiPricingPlan.php  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/CTMobi/Wubtitle-Wordpress-Plugin/pullRequest?prid=5959027&bid=19536256)

**Review by HyprCoder — APPROVED** [review]

**Pull request merged**